src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/AppealSubmitFeeHandler.java

```diff
@@ -61,30 +65,44 @@ public boolean canHandle(
 
         AppealType appealType = asylumCase.read(APPEAL_TYPE, AppealType.class)
             .orElseThrow(() -> new IllegalStateException("AppealType is not present"));
+        asylumCase.write(FEE_PAYMENT_APPEAL_TYPE, "No");
 
         if (appealType.equals(AppealType.EA)
             || appealType.equals(AppealType.HU)
             || appealType.equals(AppealType.PA)) {
-            if (!isFeeExists(FeeType.ORAL_FEE)) {
+            if (!isFeeExists(FeeType.FEE_WITH_HEARING) || (!isFeeExists(FeeType.FEE_WITHOUT_HEARING))) {
                 PreSubmitCallbackResponse<AsylumCase> response =
                     new PreSubmitCallbackResponse<>(callback.getCaseDetails().getCaseData());
                 response.addErrors(Collections.singleton("Cannot retrieve the fee from fees-register."));
 
                 return response;
             }
 
-            asylumCase.write(APPEAL_FEE_DESC,
-                "The fee for this type of appeal with a hearing is £" + fee.getCalculatedAmount());
-            asylumCase.write(ORAL_FEE_AMOUNT_FOR_DISPLAY, "£" + fee.getCalculatedAmount());
+            asylumCase.write(
+                APPEAL_FEE_HEARING_DESC,
+                "The fee for this type of appeal with a hearing is £" + feeHearing.getCalculatedAmount());
+            asylumCase.write(FEE_HEARING_AMOUNT_FOR_DISPLAY, "£" + feeHearing.getCalculatedAmount());
+            asylumCase.write(
+                APPEAL_FEE_WITHOUT_HEARING_DESC,
+                "The fee for this type of appeal with a hearing is £" + feeWithoutHearing.getCalculatedAmount());
+            asylumCase.write(FEE_WITHOUT_HEARING_AMOUNT_FOR_DISPLAY, "£" + feeWithoutHearing.getCalculatedAmount());
+
             asylumCase.write(PAYMENT_STATUS, "Payment due");
+            asylumCase.write(FEE_PAYMENT_APPEAL_TYPE, "Yes");
         }
 
         return new PreSubmitCallbackResponse<>(asylumCase);
     }
 
     private boolean isFeeExists(FeeType feeType) {
 
-        fee = feeService.getFee(feeType);
-        return fee != null;
+        if (feeType.getValue().equals(FeeType.FEE_WITH_HEARING.getValue())) {
+            feeHearing = feeService.getFee(feeType);
```

Comment:
This assigment is not thread-safe because we use global object field `feeHearing` and `feeWithoutHearing`.
I would rather use local method variable instead
I think `isFeeExists` method is not needed you can put below logic to main `handle` method

Reply:
Used local method variable to get hearing fee. Removed isFeeExists method.